ivi-layermanagement-api/test/ilm_check_env.cpp

```diff
@@ -0,0 +1,190 @@
+    bool mCheck;
+
+    static constexpr uint32_t IVI_CONTROLLER_VERSION{1U};
+    static constexpr uint32_t IVI_INPUT_VERSION{2U};
+
+    static TestEnvChecking *GetInstance();
+    ~TestEnvChecking();
+
+  private:
+    static void RegistryHandleGlobal(
+        void *apData,
+        struct wl_registry *apRegistry,
+        uint32_t aId,
+        const char *acpInterface,
+        uint32_t aVersion);
+
+    static constexpr struct wl_registry_listener mRegistryListener = {
+      RegistryHandleGlobal,
+      nullptr
+    };
+
+    static TestEnvChecking *mpInstance;
+    TestEnvChecking();
+};
+
+TestEnvChecking* TestEnvChecking::mpInstance{nullptr};
+
+TestEnvChecking::TestEnvChecking():
+    mpDisplay(nullptr),
+    mpRegistry(nullptr),
+    mpController(nullptr),
+    mpInputController(nullptr),
+    mCheck(true)
+{
+  int ret;
+
+  mpDisplay = wl_display_connect(nullptr);
+  if (mpDisplay == nullptr) {
+    printf("ERROR: Failed to wl_display_connect()\n");
+    mCheck = false;
+    return;
+  }
+
+  mpRegistry = wl_display_get_registry(mpDisplay);
+  if (mpRegistry == nullptr) {
+    printf("ERROR: Failed to wl_display_get_registry()\n");
+    mCheck = false;
+    return;
+  }
+
+  ret = wl_registry_add_listener(mpRegistry, &mRegistryListener, this);
+  if (ret < 0) {
+    printf("ERROR: Failed to wl_registry_add_listener()\n");
+    mCheck = false;
+    return;
+  }
+
+  ret = wl_display_roundtrip(mpDisplay);
+  if (ret < 0) {
+    printf("ERROR: Failed to wl_display_roundtrip()\n");
+    mCheck = false;
+    return;
+  }
+
+  if ((mpController == nullptr) || (mpInputController == nullptr)) {
+    printf("ERROR: can't get the ivi_wm (%lX) or ivi_input (%lX) protocols from compositor.\n",
+        reinterpret_cast<uintptr_t>(mpController),
+        reinterpret_cast<uintptr_t>(mpInputController));
+    mCheck = false;
+    return;
+  }
+}
+
+TestEnvChecking::~TestEnvChecking()
+{
+  if (!mpInputController) {
+    ivi_input_destroy(mpInputController);
+  }
+
+  if (!mpController) {
+    ivi_wm_destroy(mpController);
+  }
+
+  if (!mpRegistry) {
+    wl_registry_destroy(mpRegistry);
+  }
+
+  if (!mpDisplay) {
+    wl_display_disconnect(mpDisplay);
+  }
+}
+
+void TestEnvChecking::RegistryHandleGlobal(
+    void *apData,
+    struct wl_registry *apRegistry,
+    uint32_t aId,
+    const char *acpInterface,
+    uint32_t aVersion)
+{
+  TestEnvChecking *lpTestEnvChecking = reinterpret_cast<TestEnvChecking*>(apData);
+
+  if (0 == strcmp(acpInterface, "ivi_wm")) {
+    if (IVI_CONTROLLER_VERSION != aVersion) {
+      printf("ERROR: unexpected ivi_wm version from server. Expected: %d, got: %d\n",
+          IVI_CONTROLLER_VERSION,
+          aVersion);
+      lpTestEnvChecking->mCheck = false;
+      return;
+    }
+
+    lpTestEnvChecking->mpController = 
+        reinterpret_cast<struct ivi_wm*>(wl_registry_bind(
+        apRegistry,
+        aId,
+        &ivi_wm_interface,
+        IVI_CONTROLLER_VERSION));
```

Comment:
this looks like to much line breaks :-), proposal:
    lpTestEnvChecking->mpController = reinterpret_cast<struct ivi_wm*>
	           (wl_registry_bind( apRegistry, aId, &ivi_wm_interface, IVI_CONTROLLER_VERSION));

Reply:
Thank Eugen, I just updated following your suggestion